server/broadcast.go

```diff
@@ -377,6 +385,7 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 	}
 	res, err := SubmitSegment(sess, seg, nonce)
 	if err != nil || res == nil {
+		cxn.sessManager.sus.suspend(sess.OrchestratorInfo.GetTranscoder(), cxn.sessManager.poolSize/cxn.sessManager.numOrchs)
```

Comment:
Taking a look at `transcodeSegment()`, there are a few other places where we remove an O from the working set:

- When a direct upload to O fails
- When ticket params validation fails with an error that is not `ErrTicketParamsExpired`
- When an attempt to refresh the ticket params for an O fails
- When an attempt to download from O fails
- When we encounter a retryable verification error (tampering was detected, but could be a false negative)

We don't want to suspend an O for the last case (since it could be a false negative [1]), but we could suspend Os for the other cases by adding the suspension logic into `removeSession()`. In order to avoid suspending an O for the last case we could replace the [removeSession()](https://github.com/livepeer/go-livepeer/blob/ce63d6c5e7ef9da33c05b55fed7b573f0cf4adf4/server/broadcast.go#L546) call with `deleteSession()` (naming?) that does not include suspension logic and *only* removes the session from sessMap.

[1] Side note: We still need to add in logic for re-adding those Os back to the working set in that scenario.

Reply:
I'd rather stick to duplicating 
```
		cxn.sessManager.sus.suspend(sess.OrchestratorInfo.GetTranscoder(), cxn.sessManager.poolSize/cxn.sessManager.numOrchs)
```

I think suspension of an orchestrator and removal of a session are different things,  and combining this in a method could lead to confusion. For the similar reason I would also think that eg. keeping `removeSession` and introducing `suspendAndRemoveSession` wouldn't necessarily be better. 

So in my opinion, duplicate but perhaps we can introduce a wrapper method
`bsm.suspendOrchestrator(*BroadcastSession)` for brevity

Reply:
fixed in d313eb53

Created new tests for the upload and download cases